src/panels/lovelace/entity-rows/types.ts

```diff
@@ -28,8 +28,8 @@ export interface WeblinkConfig {
   icon?: string;
   url: string;
 }
-export interface CallServiceConfig extends EntityConfig {
-  type: "call-service";
+export interface ButtonRowConfig extends EntityConfig {
+  type: "button" | "call-service";
   action_name?: string;
```

Comment:
If we are creating a new type, shouldn't we align the option with the button card? So add the default actions config instead of this?

Reply:
We could have the `call-service` row config map to `button` config:
```ts
{tap_action: { action: "call-service", service: config.service, service_data: service_data}  }
```

Reply:
yes, that's a good idea

Reply:
So was about to start on this and realized that we already have actions for rows...but they are for the icon/name already. So we'd need to differentiate that these actions are for the button vs the "row". Thoughts?
```yaml
button_action:
  tap_action:
  ...
```

Reply:
That sucks, getting complicated now.
I think that is our only option?

Reply:
Why differentiate them? If the Tap Action for either row then the Row and the button do the same action. I don't see the need to do two different user defined actions. Tapping the Row should show more info and the button should do the action

Reply:
Yeah that's how a scene row works for instance.

Reply:
Yea. I think that is a better option than confusing the configuration with the proposed config above

Reply:
I agree, somewhat. We can make this work with existing actions, but actions on the icon/name are set to 'none'

Reply:
I feel like its more of an aesthetics thing for users. It's not intuitive that clicking the icon or name would perform an action. I would think it would bring up more info and the button is the only thing that performs the action.

Reply:
I spoke to this without actually looking at the code base, there are already no actions on the icon/name so bram's initial suggestion was perfect all along :)